core/iwasm/common/wasm_c_api.c

```diff
@@ -4835,7 +4874,7 @@ wasm_instance_new_with_args(wasm_store_t *store, const wasm_module_t *module,
     }
 
     /* fill in c-api func import list */
-    for (i = 0; i < import_count; i++) {
+    for (i = 0; i < imports->num_elems; i++) {
```

Comment:
imports might be NULL if the wasm app hasn't imports. And L4802 checks whether imports is NULL: `if (imports && imports->num_elems)`